src/gui/editors/PianoRoll.cpp

```diff
@@ -1930,8 +1933,9 @@ void PianoRoll::mousePressEvent(QMouseEvent * me )
 
 				Engine::getSong()->setModified();
 			}
-			else if( ( me->buttons() == Qt::RightButton &&
-							m_editMode == EditMode::Draw ) ||
+			else if ((me->buttons() == Qt::RightButton &&
+							m_editMode == EditMode::Draw &&
+							(!isSelection() || it != notes.rend())) ||
 					m_editMode == EditMode::Erase )
```

Comment:
You can fix style here as it's a continuation of the previous line

Reply:
Line 1939

Reply:
like in the Qt example?:
```
 // Correct
 if (longExpression
     + otherLongExpression
     + otherOtherLongExpression) {
 }
```

Reply:
Yup

Reply:
Also, you need to indent the line properly

Reply:
so like this(?):
```
else if ((me->buttons() == Qt::RightButton
						&& m_editMode == EditMode::Draw
						&& (!isSelection() || it != notes.rend()))
						|| m_editMode == EditMode::Erase )
```

Reply:
Remove the space at the end

Reply:
ok

Reply:
is line 1952 correctly formatted?

Reply:
There's some indentation issues there too